presto-main/src/main/java/com/facebook/presto/execution/CreateTableTask.java

```diff
@@ -163,16 +163,15 @@ else if (element instanceof LikeClause) {
         Map<String, Object> finalProperties = combineProperties(statement.getProperties().keySet(), properties, inheritedProperties);
 
         ConnectorTableMetadata tableMetadata = new ConnectorTableMetadata(tableName.asSchemaTableName(), ImmutableList.copyOf(columns.values()), finalProperties, statement.getComment());
-
+        // try/catch is needed when connectors don't handle the ignoreExisting flag, which is optional.
         try {
-            metadata.createTable(session, tableName.getCatalogName(), tableMetadata);
+            metadata.createTable(session, tableName.getCatalogName(), tableMetadata, statement.isNotExists());
```

Comment:
We need to make sure every commit compiles and passes tests (so you can bisect later). Squashing this commit into the previous one is probably easiest.